### PR TITLE
[6/N] Fixes clang-tidy warnings in c10/{core,util}/*.h

### PR DESCRIPTION
**Pull request by cyyever**
This PR continues to fix clang-tidy warnings for headers in c10/core and c10/util.

### CODE REVIEW

File: c10/core/impl/InlineEvent.h [hunk 1]
```diff
@@ -22,15 +22,20 @@ struct InlineEvent final {
 
   // Move constructor and move assignment operator
   InlineEvent(InlineEvent&& other) noexcept
-      : InlineEvent(other.device_type_, other.flag_) {
-    swap(std::move(other));
+      : event_(other.event_),
```
[the remark this answers is not in the document]
Comment (cyyever) [reply c10/core/impl/InlineEvent.h:25]: No, std::exchange has to assign nullptr to other.event, but here is for member initialisation.

[the remark this answers is not in the document]
Comment (cyyever) [reply c10/core/impl/InlineEvent.h:25]: We can use event(std::exchange(nullptr,other.event)), but that is a too clever idea to understand.

File: c10/util/complex.h [hunk 2]
```diff
@@ -332,19 +330,19 @@ struct alignas(sizeof(T) * 2) complex {
 
 namespace complex_literals {
 
-constexpr complex<float> operator"" _if(long double imag) {
+constexpr complex<float> operator""_if(long double imag) {
```
Comment (Skylion007) [inline c10/util/complex.h:333]: Is this covered under the preview of the clang-format lintrunner? If not, we should apply it so we only have to do this once.

File: c10/core/impl/SizesAndStrides.h [hunk 3]
```diff
@@ -297,7 +297,7 @@ class C10_API SizesAndStrides {
     memcpy(outOfLineStorage_, rhs.outOfLineStorage_, storageBytes(rhs.size_));
```
Comment (Skylion007) [inline c10/core/impl/SizesAndStrides.h:297]:
```suggestion
    std::memcpy(outOfLineStorage_, rhs.outOfLineStorage_, storageBytes(rhs.size_));
```
while we are here.

**Review by Skylion007 — APPROVED** [review]

File: c10/core/impl/SizesAndStrides.h (hunk 3 above)
Comment (Skylion007) [reply c10/core/impl/SizesAndStrides.h:297]: Actually, we should probably do the whole file here if we do it, feel free to disregard or do a separate PR.

File: c10/util/complex.h (hunk 2 above)
Comment (cyyever) [reply c10/util/complex.h:333]: This is a clang++ warning.
```
error: identifier '_id' preceded by whitespace in a literal operator declaration is deprecated [clang-diagnostic-deprecated-literal-operator,-warnings-as-errors]
```
Clang-format may not support fixing it.

File: c10/core/impl/SizesAndStrides.h (hunk 3 above)
Comment (cyyever) [reply c10/core/impl/SizesAndStrides.h:297]: whole file?

**Pull request closed without merging**